manifests/source.pp

```diff
@@ -109,14 +90,19 @@
       }
       $_key = merge($::apt::params::source_key_defaults, $_deprecated_key, $key)
     } else {
-      validate_string($key)
+      validate_legacy(String, 'validate_string', $key)
       $_key = merge( { 'id' => $key }, $_deprecated_key)
     }
   }
 
+  $header = epp('apt/_header.epp')
+
+  $sourcelist = epp('apt/source.list.epp', { 'comment' => $comment, 'includes' => $includes, 'architecture' => $architecture, 'allow_unsigned' => $_allow_unsigned, 'location' => $location, 'release' => $_release, 'repos' => $repos })
```

Comment:
This should probably be wrapped like your other epp hashes, or alternately my preferred form for hashes in puppet:
```puppet
  $sourcelist = epp('apt/source.list.epp', {
    'comment'        => $comment,
    'includes'       => $includes,
    'architecture'   => $architecture,
    'allow_unsigned' => $_allow_unsigned,
    'location'       => $location,
    'release'        => $_release,
    'repos'          => $repos,
  })
```